hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,915 @@
+        activity_type_id = data.get("activity_type_id")  # ID of the activity type
+        summary = data.get("summary", "")  # Summary text of the activity (optional)
+        date_deadline = data.get("date_deadline")  # Deadline date for the activity
+        note = data.get("note", "")  # Note for the activity (optional)
+        user_id = data.get("user_id", request.env.uid)  # User assigned to the activity
+
+        # Create the scheduled activity
+        activity = request.env["mail.activity"].create(
+            {
+                "activity_type_id": activity_type_id,
+                "note": note,
+                "date_deadline": date_deadline,
+                "res_model_id": request.env["ir.model"]._get("sale.order").id,
+                "res_id": sale_order_id,
+                "user_id": user_id,
+                "summary": summary,
+            }
+        )
+        logger.info(
+            "Scheduled activity created with ID %s for sale order %s",
+            activity.id,
+            sale_order_id,
+        )
+
+        return {
+            "message": f"Scheduled activity created with ID: {activity.id} for sale order {sale_order_id}."
+        }
+
+    @route(
+        "/create_schedule_activity_invoice", methods=["POST"], type="json", auth="user"
+    )
+    def create_schedule_activity_invoice(self):
+        """
+        Creates a scheduled activity associated with an existing invoice.
+
+        This function creates a scheduled activity in Odoo related to a specific invoice,
+        using the details provided in the JSON request body.
+
+        JSON request body:
+            - invoice_id (int): ID of the existing invoice.
+            - activity_type_id (int): ID of the activity type.
+            - summary (str, optional): Summary text of the activity.
+            - date_deadline (str): Deadline date for the activity in 'YYYY-MM-DD' format.
+            - note (str, optional): Note for the activity.
+            - user_id (int, optional): ID of the user assigned to the activity. If not provided, the current user's ID is used.
+
+        JSON response:
+            - message (str): Confirmation message with the ID of the scheduled activity and the invoice ID.
+
+        Returns:
+            dict: Dictionary with a confirmation message.
+        """
+        data = request.jsonrequest
+        invoice_id = data.get("invoice_id")  # ID of the existing invoice
+        activity_type_id = data.get("activity_type_id")  # ID of the activity type
+        summary = data.get("summary", "")  # Summary text of the activity (optional)
+        date_deadline = data.get("date_deadline")  # Deadline date for the activity
+        note = data.get("note", "")  # Note for the activity (optional)
+        user_id = data.get("user_id", request.env.uid)  # User assigned to the activity
+
+        # Create the scheduled activity
+        activity = request.env["mail.activity"].create(
```

Comment:
no man, use invoice.activity_schedule
https://github.com/odoo/odoo/blob/ab01f91913f32e11cc20287f719933849f0949eb/addons/mail/wizard/mail_activity_schedule.py#L216

Reply:
@JAntonioSalas 
missing this

Reply:
I already pushed this change and the others @hugho-ad